soql-standalone-parser/src/main/jflex/SoQLLexer.flex

```diff
@@ -126,6 +131,8 @@ QuotedSystemIdentifier = ":" "@"? ("-" | [:jletterdigit:])+
   {SystemIdentifier} { return token(new SystemIdentifier(yytext(), false)); }
   "`" / ":" { stringStart = pos(); yybegin(QUOTEDSYSTEMIDENTIFIER); }
 
+  {TableIdentifier} { return token(new TableIdentifier(yytext())); }
```

Comment:
If you don't want the `@` to be a part of the name (I see it gets stripped off in the parser), you can `.substring(1)` the yytext here.